internal/utils.go

```diff
@@ -310,6 +316,23 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]interface{
 	return policyTemplates, nil
 }
 
+// setGatekeeperEnforcementAction function override gatekeeper.constraint.enforcementAction
+func setGatekeeperEnforcementAction(manifest map[string]interface{}, enforcementAction string) error {
+	apiVersion, found, err := unstructured.NestedString(manifest, "apiVersion")
+	if err != nil {
+		return fmt.Errorf("getting apiVersion has an error %w", err)
+	}
+
+	if found && strings.HasPrefix(apiVersion, "constraints.gatekeeper.sh") && enforcementAction != "" {
+		err := unstructured.SetNestedField(manifest, enforcementAction, "spec", "enforcementAction")
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
```suggestion
	if enforcementAction == "" {
		return nil
	}

	apiVersion, _, _ := unstructured.NestedString(manifest, "apiVersion")

	if strings.HasPrefix(apiVersion, "constraints.gatekeeper.sh") {
		err := unstructured.SetNestedField(manifest, enforcementAction, "spec", "enforcementAction")
		if err != nil {
			return err
		}
	}

	return nil
```

Reply:
I think it would be more safe that we have error check here 
`apiVersion, _, err := unstructured.NestedString(manifest, "apiVersion")
if err != nil {
			return err
		}
`

Reply:
@yiraeChristineKim that's fine but I kind of wanted to avoid validating the actual manifest in the Policy Generator and let the policy controllers handle it.